File: c++/1182.cpp
#include <iostream>
#include <vector>

using namespace std;

int n, s, answer;
vector<int> v;

void dfs(int sum, int idx) {
  if (idx != 0 && sum == s) answer++;
  for (int i = idx; i < n; i++) {
    sum += v[i];
    dfs(sum, i + 1);
    sum -= v[i];
  }
}

int main() {
  // freopen("example.txt", "r", stdin);

  cin >> n >> s;

  for (int i = 0; i < n; i++) {
    int a;
    cin >> a;
    v.push_back(a);
  }
  dfs(0, 0);
  cout << answer;

  return 0;
}

File: js/1182.js
const fs = require('fs');

const inputs = fs.readFileSync('/dev/stdin').toString().trim().split('\n');

const [n, s] = inputs[0].split(' ').map(Number);
const arr = inputs[1].split(' ').map(Number);

let answer = 0;

dfs(0, 0);
console.log(answer);

function dfs(sum, idx) {
  if (idx !== 0 && sum === s) answer++;
  for (let i = idx; i < n; i++) {
    sum += arr[i];
    dfs(sum, i + 1);
    sum -= arr[i];
  }
}
